### PR TITLE
RI-15 - Related content on search

### PR DESCRIPTION
**Pull request by ledjay**
Ticket : 
https://linear.app/refugiesinfo/issue/RI-15/rr-section-autre-fiches

- finishes the related content rules and filters on search
- adds the markup on the search page dom

### CODE REVIEW

File: apps/client/src/components/Pages/recherche/SearchResults/SearchResults.tsx [hunk 1]
```diff
@@ -152,6 +150,24 @@ const SearchResults = (props: Props) => {
             </div>
           </>
         )}
+        {filteredResults.suggestions.length > 0 && (
```
Comment (kaaloo) [inline apps/client/src/components/Pages/recherche/SearchResults/SearchResults.tsx]: I think you need to call the [getDisplayRuleForQuery](https://github.com/refugies-info/karfur/blob/af6d5d147ea35aef9864c4961357d1547fa75c6c/apps/client/src/lib/recherche/queryContents.ts#L39) function to know if the suggestions should be displayed or not.  For instance:

```
getDisplayRuleForQuery(query, "suggestions")
```

on top of checking if the suggestions item is field.

File: apps/client/src/lib/recherche/queryContents.ts [hunk 2]
```diff
@@ -78,9 +79,59 @@ const filterDispositifs = (
     .filter((dispositif) => filterByLanguage(dispositif, query.language))
     .filter((dispositif) => filterByPublic(dispositif, query.public))
     .filter((dispositif) => filterByStatus(dispositif, query.status));
+
   return rule?.sortFunction ? [...filteredDispositifs].sort((a, b) => rule.sortFunction(a, b)) : filteredDispositifs;
 };
 
+const filterSuggestions = (
```
Comment (kaaloo) [inline apps/client/src/lib/recherche/queryContents.ts:86]: Wow, tough filtering rule, crossing fingers !

File: apps/client/src/components/Pages/recherche/SearchResults/SearchResults.tsx (hunk 1 above)
Comment (kaaloo) [reply apps/client/src/components/Pages/recherche/SearchResults/SearchResults.tsx]: For instance the suggestions shouldn't be shown in the case where there is a location and keywords:

image

**Review by kaaloo — CHANGES_REQUESTED** [review]
Very cool stuff !  Just need to make use of the getDisplayRuleForQuery function on top of your current processing (see comments)

File: apps/client/src/lib/recherche/queryContents.ts (hunk 2 above)
Comment (ledjay) [reply apps/client/src/lib/recherche/queryContents.ts:86]: Indeed!

File: apps/client/src/components/Pages/recherche/SearchResults/SearchResults.tsx (hunk 1 above)
Comment (ledjay) [reply apps/client/src/components/Pages/recherche/SearchResults/SearchResults.tsx]: Thanks for the feedback. I just pushed the fix, is it what you had in mind?

**Review by kaaloo — APPROVED** [review]
Looks good yes !  I would wrap that in a useMemo though WDYT ?  Your call :)

**Pull request merged**